packages/amplify_core/lib/amplify_hub.dart

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import './amplify_core.dart';
+class AmplifyHub {
+
+  Map<HubChannel, Stream> availableStreams = {};
+  
+  /// Expose listen method which instantiates new StreamController listening to one or more availableStreams
+  StreamSubscription listen(List<HubChannel> channels, @required dynamic listener) {
```

Comment:
I don't need any special syntax here to just listen to _one_ channel, right?

Also, why `dynamic`? Don't you want some stable callback interface type, here?

What's the listener actually used for? Don't consumers receive events from the Stream, not from this listener?

Reply:
> I don't need any special syntax here to just listen to one channel, right?

Assuming that "I" refers the customer, then no.  

> Also, why dynamic? Don't you want some stable callback interface type, here?

I've tightened this up slightly by user a function typedef that accepts one parameter, but being any more specific than that would require the customer to import the HubEvent type on their end, and also seems to create problems due to the fact that the parameters in these listeners are contravariant. 

> What's the listener actually used for? Don't consumers receive events from the Stream, not from this listener?

We are essentially re-broadcasting events from the various HubChannel streams onto a single stream which we create for the customer, and then call 'listen' with the listener they've given us.